internal/server/evaluation/evaluator.go

```diff
@@ -0,0 +1,430 @@
+package evaluation
+
+import (
+	"context"
+	"hash/crc32"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	errs "go.flipt.io/flipt/errors"
+	"go.flipt.io/flipt/internal/server/metrics"
+	"go.flipt.io/flipt/internal/storage"
+	"go.flipt.io/flipt/rpc/flipt"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"go.uber.org/zap"
+)
+
+// EvaluationStorer is the minimal abstraction for interacting with the storage layer for evaluation.
+type EvaluationStorer interface {
+	GetFlag(ctx context.Context, namespaceKey, key string) (*flipt.Flag, error)
+	GetEvaluationRules(ctx context.Context, namespaceKey string, flagKey string) ([]*storage.EvaluationRule, error)
+	GetEvaluationDistributions(ctx context.Context, ruleID string) ([]*storage.EvaluationDistribution, error)
+}
+
+// Evaluator is responsible for legacy evaluations.
+type Evaluator struct {
+	logger *zap.Logger
+	store  EvaluationStorer
+}
+
+// NewEvaluator is the constructor for an Evaluator.
+func NewEvaluator(logger *zap.Logger, store EvaluationStorer) *Evaluator {
+	return &Evaluator{
+		logger: logger,
+		store:  store,
+	}
+}
+
+func (e *Evaluator) Evaluate(ctx context.Context, flag *flipt.Flag, r *flipt.EvaluationRequest) (resp *flipt.EvaluationResponse, err error) {
+	// Flag should be of variant type by default. However, for maximum backwards compatibility
+	// on this layer, we will check against the integer value of the flag type.
+	if int(flag.Type) != 0 || flag.Type != flipt.FlagType_VARIANT_FLAG_TYPE {
```

Comment:
nit: i think this is redundant since `flipt.FlagType_VARIANT_FLAG_TYPE == 0` but also doesn't hurt anything